src/secret.rs

```diff
@@ -0,0 +1,311 @@
+use fuel_types::Bytes32;
+
+use core::fmt;
+use core::ops::Deref;
+
+/// Signature secret key
+#[derive(Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct SecretKey(Bytes32);
+
+impl SecretKey {
+    /// Memory length of the type
+    pub const LEN: usize = Bytes32::LEN;
+
+    /// Add a conversion from arbitrary slices into owned
+    ///
+    /// # Safety
+    ///
+    /// There is no guarantee the provided bytes will fit the field. The field
+    /// security can be checked with [`SecretKey::is_in_field`].
+    pub unsafe fn from_bytes_unchecked(bytes: [u8; Self::LEN]) -> Self {
+        Self(bytes.into())
+    }
+
+    /// Add a conversion from arbitrary slices into owned
+    ///
+    /// # Safety
+    ///
+    /// This function will not panic if the length of the slice is smaller than
+    /// `Self::LEN`. Instead, it will cause undefined behavior and read random
+    /// disowned bytes.
+    ///
+    /// There is no guarantee the provided bytes will fit the field.
+    pub unsafe fn from_slice_unchecked(bytes: &[u8]) -> Self {
+        Self(Bytes32::from_slice_unchecked(bytes))
+    }
+
+    /// Copy-free reference cast
+    ///
+    /// # Safety
+    ///
+    /// This function will not panic if the length of the slice is smaller than
+    /// `Self::LEN`. Instead, it will cause undefined behavior and read random
+    /// disowned bytes.
+    ///
+    /// There is no guarantee the provided bytes will fit the field.
+    pub unsafe fn as_ref_unchecked(bytes: &[u8]) -> &Self {
+        // The interpreter will frequently make references to keys and values using
+        // logically checked slices.
+        //
+        // This function will save unnecessary copy to owned slices for the interpreter
+        // access
```

Comment:
nit: The wording is  here confusing me, if I understand correctly, "avoid" instead of "save" might be clearer

Reply:
https://github.com/FuelLabs/fuel-crypto/pull/1/commits/1e596d7057581b6cce695ea847a2845f689160ef